hs_core/migrations/0083_auto_20240702_2038.py

```diff
@@ -10,16 +10,38 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        migrations.RemoveField(
+        # make a field nullable
+        migrations.AlterField(
             model_name='baseresource',
             name='resource_federation_path',
+            field=models.CharField(blank=True, default='', max_length=100, null=True),
         ),
-        migrations.AlterIndexTogether(
-            name='resourcefile',
-            index_together={('object_id', 'resource_file')},
+        migrations.SeparateDatabaseAndState(
+            state_operations=[
+                migrations.RemoveField(
+                    model_name='baseresource',
+                    name='resource_federation_path',
+                ),
+                migrations.AlterIndexTogether(
+                    name='resourcefile',
+                    index_together={('object_id', 'resource_file')},
+                ),
+                migrations.RemoveField(
+                    model_name='resourcefile',
+                    name='fed_resource_file',
+                ),
+            ],
+            # Operations to be performed later on the database.
+            database_operations=[
+                # migrations.RunSQL(
```

Comment:
Do we want to keep this commented code?

Reply:
I vote yes, I thought it was helpful to see the operations that would be run later.

Reply:
really nice question. I would think you're right, but interestingly, if I checkout 2.15.1 (ie this commit f7f200742039cdd9cad07bcc341c8efee9e5c4cc) and run `./hsctl managepy sqlmigrate hs_core 0083_auto_20240702_2038` here is what it shows:
```sh
➜ ./hsctl managepy sqlmigrate hs_core 0083_auto_20240702_2038
MANAGEPY:
  - docker exec -u hydro-service -ti hydroshare python manage.py sqlmigrate hs_core 0083_auto_20240702_2038
/usr/local/lib/python3.9/site-packages/mezzanine/utils/timezone.py:13: PytzUsageWarning: The zone attribute is specific to pytz's interface; please migrate to a new time zone provider. For more details on how to do so, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  zone_name = tzlocal.get_localzone().zone
/usr/local/lib/python3.9/site-packages/mezzanine/utils/conf.py:67: UserWarning: TIME_ZONE setting is not set, using closest match: Etc/UTC
  warn("TIME_ZONE setting is not set, using closest match: %s" % tz)
BEGIN;
--
-- Remove field resource_federation_path from baseresource
--
ALTER TABLE "hs_core_genericresource" DROP COLUMN "resource_federation_path" CASCADE;
--
-- Alter index_together for resourcefile (1 constraint(s))
--
DROP INDEX IF EXISTS "hs_core_resourcefile_object_id_fed_resource_file_15f4618e_idx";
--
-- Remove field fed_resource_file from resourcefile
--
ALTER TABLE "hs_core_resourcefile" DROP COLUMN "fed_resource_file" CASCADE;
COMMIT;
```

Reply:
I guess the reason is because of the meta on our `baseresource` class
https://github.com/hydroshare/hydroshare/blob/2.15/hs_core/models.py#L3886

So when we execute raw sql queries they have to use the db_table name instead of the model name

Reply:
You are right, we have overridden the default table name for the BaseResource.